gitlab/resource_gitlab_group_share_group.go

```diff
@@ -105,7 +105,10 @@ func resourceGitlabGroupShareGroupRead(d *schema.ResourceData, meta interface{})
 			d.Set("group_id", groupId)
 			d.Set("share_group_id", sharedGroup.GroupID)
 			d.Set("group_access", accessLevel[convertedAccessLevel])
-			d.Set("expires_at", sharedGroup.ExpiresAt.String())
+
+			if sharedGroup.ExpiresAt != nil {
+				d.Set("expires_at", sharedGroup.ExpiresAt.String())
+			}
```

Comment:
So the behavior will be to not pass `expires_at` to the API if it is `nil`. Makes sense, the only concern would be if `nil` had a special meaning, like removing an expiration, but since we don't have an update endpoint, and `ForceNew` is set on all attributes, I think this is fine the way you have it. Great work 👍

Reply:
Hehe I have the same nitpick, but it's more of a semantic thing. Since the `expires_at` field cannot be modified upstream, so it doesn't make a difference in this case. Though, in a typical read function, you'd want to set each attribute which is returned from the GitLab API, i.e. set "expires_at" to "" in the case where ExpiresAt is nil.

Reply:
I've updated to 

```
if sharedGroup.ExpiresAt == nil {
				d.Set("expires_at", "")
} else {
				d.Set("expires_at", sharedGroup.ExpiresAt.String())
}
```

I thought the empty string would fail the `validateDateFunc` but it seems to be ok now 😅

Reply:
Looks great!